assets/js/atomic/blocks/product/add-to-cart/shared/add-to-cart-button.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import { __, _n, sprintf } from '@wordpress/i18n';
+import Button from '@woocommerce/base-components/button';
+import { Icon, done as doneIcon } from '@woocommerce/icons';
+import { useState } from '@wordpress/element';
+import { useAddToCartFormContext } from '@woocommerce/base-context';
+
+/**
+ * Add to Cart Form Qty + Button Block Component.
+ */
+const AddToCartButton = () => {
+	const {
+		showFormElements,
+		product,
+		quantityInCart,
+		formDisabled,
+		formSubmitting,
+		onSubmit,
+	} = useAddToCartFormContext();
+
+	const {
+		is_purchasable: isPurchasable = true,
+		has_options: hasOptions,
+		add_to_cart: addToCartButtonData = {
+			url: '',
+			text: '',
+		},
+	} = product;
+
+	if ( ( showFormElements || ! hasOptions ) && isPurchasable ) {
```

Comment:
I'm a bit confused on why this is conditional on `showFormElements || ! hasOptions`. I'm guessing this is appropriate, can you clarify so I understand? Maybe add a comment if you think that's useful.

`showFormElements` - does this component need to be aware, or could the higher-level component (`Block`) simply not render `AddToCartButton` when form elements are toggled off? 

`hasOptions` - when a product doesn't have options, why would that force the button to render?

Reply:
Added a comment.

> // If we are showing form elements, OR if the product has no additional form options, we can show
> // a functional direct add to cart button, provided that the product is purchasable. 
> // No link is required to the full form under these circumstances.